signature/policy_eval.go

```diff
@@ -14,10 +14,20 @@ import (
 )
 
 // PolicyRequirementError is an explanatory text for rejecting a signature or an image.
-type PolicyRequirementError string
+type PolicyRequirementError struct {
+	message string
+}
+
+// NewPolicyRequirementError is used to create a new PolicyRequirementError
+func NewPolicyRequirementError(message string) *PolicyRequirementError {
```

Comment:
Oh, and this can be a value type (not a pointer). A single-member struct is hopefully just as cheap as the member.

That way the `PolicyRequirementError` can be used in callers directly without having to know about the pointer.